googletest/include/gtest/internal/gtest-port.h

```diff
@@ -603,6 +603,13 @@ struct _RTL_CRITICAL_SECTION;
     || GTEST_OS_QNX || GTEST_OS_FREEBSD || GTEST_OS_NACL)
 #endif  // GTEST_HAS_PTHREAD
 
+// On MinGW, we can have both GTEST_OS_WINDOWS and GTEST_HAS_PTHREAD
+// defined, but we don't want to use MinGW's pthreads implementation, which
+// has conformance problems with some versions of the POSIX standard.
+#if GTEST_HAS_PTHREAD && GTEST_OS_WINDOWS_MINGW
+#undef GTEST_HAS_PTHREAD
```

Comment:
For reference, I believe the issue might have something to do with MinGW and its pthread implementation, as referenced in an [old bug-report](https://github.com/google/googletest/issues/298#issuecomment-125644818).

Initially the "fix" I implemented above was what made gtest compile for me using MinGW gcc 4.9.2 32bit (unfortunately I do not have the details in front of me), and should be considered a band-aid for the commit which actually introduced the original logic, but which was broken AFAIK, as it was merely negating [a single GTEST_HAS_PTHREAD](https://github.com/google/googletest/pull/608/files#diff-f21e2c91878df000cc7b8b316a89565cL1558) \#ifdef. The commit that I think might have introduced it was a634042

However I see now that cmake seems to [enable pthreads also](https://github.com/google/googletest/blob/master/googletest/cmake/internal_utils.cmake#L122).